hts.c

```diff
@@ -58,6 +58,11 @@ DEALINGS IN THE SOFTWARE.  */
 #include "htslib/kseq.h"
 #include "htslib/ksort.h"
 #include "htslib/tbx.h"
+#if defined(HAVE_EXTERNAL_LIBHTSCODECS)
```

Comment:
This works fine, but I wonder if it's the wrong method.

The traditional way would be to amend the include search path to include the htscodecs sub-module dir, so the include works using local source.  I guess the tricky bit is needing -I- or -iquote rules in order to include this dir before any system paths.  I'm not sure how portable those methods are.

Hence the above is at least robust and portable, but I wondered if you thought about the other approaches?

Reply:
It's deliberately done this way to avoid issue #347 (HTSlib should prefer its own headers when being compiled).  I doubt other solutions are very portable, and this solution is already used in source files in the `cram` subdirectory.